pkg/sql/codegen/pai/kmeans.go

```diff
@@ -20,33 +20,73 @@ import (
 	"sqlflow.org/sqlflow/pkg/database"
 	"sqlflow.org/sqlflow/pkg/ir"
 	pb "sqlflow.org/sqlflow/pkg/proto"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/attribute"
 )
 
+var kmeansAttributes = attribute.Dictionary{
+	"center_count": {attribute.Int, 3, `[default=3]
+The cluster count. range: [1, Infinity]
+`, attribute.IntLowerBoundChecker(1, true)},
+	"idx_table_name": {attribute.String, "", `
+The output table name which includes
+cluster_index column indicates the cluster result,
+distance column indicates the distance from the center and
+all the columns of input table.`, nil},
+	"excluded_columns": {attribute.String, "", `[default=""]
+excluded the special feature columns from the SELECT statment.`, nil},
+}
+
+func parseExcludedColsMap(attrs map[string]interface{}) (map[string]int, error) {
+	excludedColsMap := make(map[string]int)
+	excludedColsAttr := attrs["excluded_columns"].(string)
+	if excludedColsAttr != "" {
+		arr := strings.Split(excludedColsAttr, ",")
+		for _, e := range arr {
+			excludedColsMap[e] = 1
+		}
+	}
+	return excludedColsMap, nil
```

Comment:
It looks like the error(nil) in `return` is unnecessary.

Reply:
That makes easy to do the unit test, but I think you're right, will add it if necessary.